jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/ConfigurationPropertyValidator.java

```diff
@@ -128,5 +134,59 @@ public static ImageReference getGeneratedTargetDockerTag(
     }
   }
 
+  /**
+   * Parses a string in the form of "key1=value1,key2=value2,..." into a map.
+   *
+   * @param property the map string to parse, with entries separated by "," and key-value pairs
+   *     separated by "="
+   * @return the map of parsed values
+   */
+  public static Map<String, String> parseMapProperty(String property) {
+    // Matches key-value pairs in the form of "key=value"
+    Pattern environmentPattern = Pattern.compile("(?<name>[^=]+)=(?<value>.*)");
+    Map<String, String> result = new HashMap<>();
+
+    // Split on non-escaped commas
+    List<String> entries = parseListProperty(property);
+    for (String entry : entries) {
+      Matcher matcher = environmentPattern.matcher(entry);
+      if (!matcher.matches()) {
+        throw new IllegalArgumentException("'" + entry + "' is not a valid key-value pair");
+      }
+      result.put(matcher.group("name"), matcher.group("value"));
+    }
+    return result;
+  }
+
+  /**
+   * Parses a comma-separated string into a list. Ignores commas escaped with "\".
+   *
+   * @param property the comma-separated string
+   * @return the list of parsed values
+   */
+  public static List<String> parseListProperty(String property) {
```

Comment:
I think everywhere else we parse lists, we can just do it automatically with a json translator. But we can't really do that since the format here is slightly different. I was going off of the format I saw [in this answer](https://stackoverflow.com/questions/31620967/any-way-to-specify-a-fileset-as-command-line-parameter), and assumed commas should be escaped using `\` since `\` is such a common escape character.

Reply:
Not that this is unacceptable. Just giving more thoughts, and I kinda like I have seen this key-value parsing somewhere.

Reply:
We do have the same regex pattern for environment variables in `JsonToImageTranslator`, but in that case the list is already split up, so we only need to parse the keys/values from each entry.